src/Interpreters/InterpreterSystemQuery.cpp

```diff
@@ -374,19 +373,10 @@ BlockIO InterpreterSystemQuery::execute()
             break;
         }
         case Type::RELOAD_MODEL:
-        {
-            getContext()->checkAccess(AccessType::SYSTEM_RELOAD_MODEL);
-
-            auto & external_models_loader = system_context->getExternalModelsLoader();
-            external_models_loader.reloadModel(query.target_model);
-            break;
-        }
         case Type::RELOAD_MODELS:
         {
             getContext()->checkAccess(AccessType::SYSTEM_RELOAD_MODEL);
-
-            auto & external_models_loader = system_context->getExternalModelsLoader();
-            external_models_loader.reloadAllTriedToLoad();
+            /// SYSTEM RELOAD MODEL(S) is no longer supported. For compat reasons, it is not completely removed but retained as no-op.
```

Comment:
Fix for RBAC issue.

Reply:
I spend quite some time today to verify locally that the issue found by Nikita in the cloud (aka. reason for revert #40877) no longer happens.

Overall, I think "normal" users would not have run into the issue anyway - 1. it requires granting the [*undocumented*](https://clickhouse.com/docs/en/sql-reference/statements/grant/#grant-select) `SYSTEM RELOAD MODELS` privilege to a user/role and 2. ... which is needed only for the slightly obscure catboost usecase. My guess is that Nikitas cloud machine configuration has some history and is pretty "wild".

(Just FYI for the record: Besides `SYSTEM RELOAD MODELS`, this PR also removes view `SYSTEM.MODELS`. In theory, there could be a similar problem for SELECT privileges on this view. In practice, there isn't because SELECT privileges can be [defined on non-existing tables](https://clickhouse.com/docs/en/operations/access-rights/#access-control-properties).)

To sum up, this PR is good to merge in my POV.